### PR TITLE
Improved  TypeScript API

### PR DESCRIPTION
**Pull request by alexpmay**
* Wrote three examples, react-example, ssr-example, static-events-example
* Updated the Typescript API to sand off the rough edges I encountered. Most notably:
  * Removed requestJson. To get JSON call toJSON on an impression.
  * renamed useCasual -> useImpression
  * renamed data -> impression
* Heavily documented the examples with comments - will use as basis for the getting started guides
* Manually generated the api files using the causal/parser repo. See corresponding Causal PR. Will update npm once everything is merged

### CODE REVIEW

File: getting-started/pages/react-example.tsx [hunk 1]
```diff
@@ -1,38 +1,118 @@
...
-  const { data, loading, error } = useCausal(query, {
-    deviceId: "abc123",
-  });
+// CausalClient is the TypeScript interface to Causal
+//
+// Usually instantiation of CausalClient would be done in
+// a more global spot. In Next.js terms, often a Layout or in _app.tsx
+// It is here to show a self contained example
+//
+// By default Causal will cache results for 10 minutes. For demo purposes we
```
Comment (jpalmucci) [inline getting-started/pages/react-example.tsx:75]: Shouldn't the client cache indefinitely?

File: getting-started/pages/ssr-example.tsx [hunk 2]
```diff
@@ -1,3 +1,122 @@
...
+// create a query and also to pass data around.
+//
+// For this use case, SelectFeatures defines the type and createQuery
+// (used in gerServerSideProps) create the query
+//
+// SelectFeatures takes in the feature names to use as a generic parameter
+// i.e:  "Feature1" | "Feature2" | "Feature3"
+//
+// In VSCode, Use a double quote (") to trigger automplete within this <>
+type RatingFeatures = SelectFeatures<"RatingBox">;
```
Comment (jpalmucci) [inline getting-started/pages/ssr-example.tsx:68]: It's a bit weird that this is below it's actual usage in the file.

**Review by jpalmucci — APPROVED** [review]
minor comments

File: getting-started/pages/ssr-example.tsx (hunk 2 above)
Comment (alexpmay) [reply getting-started/pages/ssr-example.tsx:68]: Yes. I did it for the narrative flow of the comments, but I'm open to suggestions.

File: getting-started/pages/static-events-example.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+ ***/
+
+// In this example we are going to:
+// 1. Collect the data server side.
+// 2. Pass the data to the page w/o the help of Causal's types.
+// 3. use Causa's static functions on the client to signal events.
+//
+// This mimics how front end code might interact with a micro-service.
+// You might want a microservice to get data from Causal. That microservice
+// may already have a public interface, so you don't want to add new piping.
```
[the remark this answers is not in the document]
Comment (alexpmay) [reply getting-started/pages/static-events-example.tsx:18]: I'm going to leave for the moment, and update comments as I integrate with the docs.

**Pull request merged**